### DISCUSSION TITLE
Thousand separator mask for Number element

### DISCUSSION THREAD

**Discussion by DMazykin** [zauberzeug/nicegui#1802] [Ideas / Feature Requests]
It would be great to add `mask` element to allow use to define thousand separator like here: 
```html
<q-input
  filled
  v-model="price"
  label="Price with 2 decimals"
  mask="#.##"
  fill-mask="#"
  reverse-fill-mask
  hint="Mask: #.00"
  input-class="text-right"
/>
```

**Comment by rodja** [comment]
Interesting! It works with `ui.input`:

```py
    ui.input('Number', value=42000) \
        .props('mask="#.##" fill-mask="#" reverse-fill-mask hint="Mask: #.00" input-class="text-right"')
```

But not with `ui.number`. I think its due to the Quasar warning in the docs ["Mask is only available if the type is one of ‘text’ (default), ‘search’, ‘url’, ‘tel’, or ‘password’."](https://quasar.dev/vue-components/input#input-types:~:text=WARNING-,Mask%20is%20only%20available%20if%20the,is%20one%20of%20%E2%80%98text%E2%80%99%20(default)%2C%20%E2%80%98search%E2%80%99%2C%20%E2%80%98url%E2%80%99%2C%20%E2%80%98tel%E2%80%99%2C%20or%20%E2%80%98password%E2%80%99.,-Below%20are%20mask)

**Comment by DMazykin** [comment]
Thanks @rodja 
While the mask is functional with **ui.input**, it doesn't fully enhance the user experience for currency formatting. A potential solution could be to develop a new "Currency" component, inspired by the behaviour demonstrated here.: [https://vue-currency-input-v1.netlify.app/](https://vue-currency-input-v1.netlify.app/)
